lib/android-helpers.js

```diff
@@ -373,13 +373,23 @@ helpers.unlock = async function (adb) {
 
   await retryInterval(10, 1000, async () => {
     logger.debug("Screen is locked, trying to unlock");
-    await adb.startApp({
+
+    // first manually stop the unlock activity
+    await adb.forceStop('io.appium.unlock');
+
+    // then start the app twice, as once is flakey
+    let startOpts = {
       pkg: "io.appium.unlock",
       activity: ".Unlock",
       action: "android.intent.action.MAIN",
       category: "android.intent.category.LAUNCHER",
-      flags: "0x10200000"
-    });
+      flags: "0x10200000",
+      stopApp: false
+    };
+    await adb.startApp(startOpts);
+    await adb.startApp(startOpts);
+
+    // wait a moment, and check if it worked
     await sleep(1000);
```

Comment:
The whole thing is already within a `retryInterval` loop. Here we only need to check once, so I'm not sure a `retryInterval` would give us much. We could check at the very beginning of the loop, and break out if it is unlocked. But then the last time through would not have any check to see if it had worked.